frontend/src/container/CreateAlertV2/EvaluationSettings/TimeInput/TimeInput.tsx

```diff
@@ -0,0 +1,155 @@
+		`${h.padStart(2, '0')}:${m.padStart(2, '0')}:${s.padStart(2, '0')}`;
+
+	// Handle input change
+	const handleTimeChange = (
+		newHours: string,
+		newMinutes: string,
+		newSeconds: string,
+	): void => {
+		const formattedValue = formatTimeValue(newHours, newMinutes, newSeconds);
+		onChange?.(formattedValue);
+	};
+
+	// Handle hours change
+	const handleHoursChange = (e: React.ChangeEvent<HTMLInputElement>): void => {
+		const newHours = e.target.value.replace(/\D/g, '').slice(0, 2);
+		setHours(newHours);
+		handleTimeChange(newHours, minutes, seconds);
+	};
+
+	// Handle minutes change
+	const handleMinutesChange = (e: React.ChangeEvent<HTMLInputElement>): void => {
+		const newMinutes = e.target.value.replace(/\D/g, '').slice(0, 2);
+		setMinutes(newMinutes);
+		handleTimeChange(hours, newMinutes, seconds);
+	};
+
+	// Handle seconds change
+	const handleSecondsChange = (e: React.ChangeEvent<HTMLInputElement>): void => {
+		const newSeconds = e.target.value.replace(/\D/g, '').slice(0, 2);
+		setSeconds(newSeconds);
+		handleTimeChange(hours, minutes, newSeconds);
+	};
+
+	// Helper functions for field navigation
+	const getNextField = (current: string): string => {
+		switch (current) {
+			case 'hours':
+				return 'minutes';
+			case 'minutes':
+				return 'seconds';
+			default:
+				return 'hours';
+		}
+	};
+
+	const getPrevField = (current: string): string => {
+		switch (current) {
+			case 'seconds':
+				return 'minutes';
+			case 'minutes':
+				return 'hours';
+			default:
+				return 'seconds';
+		}
+	};
+
+	// Handle key navigation
+	const handleKeyDown = (
+		e: React.KeyboardEvent<HTMLInputElement>,
+		currentField: 'hours' | 'minutes' | 'seconds',
+	): void => {
+		if (e.key === 'ArrowRight' || e.key === 'Tab') {
+			e.preventDefault();
+			const nextField = document.querySelector(
+				`[data-field="${getNextField(currentField)}"]`,
+			) as HTMLInputElement;
+			nextField?.focus();
+		} else if (e.key === 'ArrowLeft') {
+			e.preventDefault();
+			const prevField = document.querySelector(
+				`[data-field="${getPrevField(currentField)}"]`,
+			) as HTMLInputElement;
+			prevField?.focus();
+		}
+	};
+
+	return (
+		<div className={`time-input-container ${className}`}>
+			<Input
+				data-field="hours"
+				value={hours}
+				onChange={handleHoursChange}
+				onKeyDown={(e): void => handleKeyDown(e, 'hours')}
+				disabled={disabled}
+				maxLength={2}
+				className="time-input-field"
+			/>
+			<span className="time-input-separator">:</span>
+			<Input
+				data-field="minutes"
+				value={minutes}
+				onChange={handleMinutesChange}
+				onKeyDown={(e): void => handleKeyDown(e, 'minutes')}
+				disabled={disabled}
+				maxLength={2}
+				className="time-input-field"
+			/>
+			<span className="time-input-separator">:</span>
+			<Input
+				data-field="seconds"
+				value={seconds}
+				onChange={handleSecondsChange}
+				onKeyDown={(e): void => handleKeyDown(e, 'seconds')}
```

Comment:
probably not needed but debounce maybe.

Reply:
Don't think debounce is required here? As we're only shuffling between the 3 inputs on key down